src/lib/net/src/packets/outgoing/chunk_and_light_data.rs

```diff
@@ -89,51 +89,47 @@ impl ChunkAndLightData {
             };
             block_light_data.push(section_block_light_data);
 
-            data.write_u16(section.block_states.non_air_blocks).await?;
+            data.write_u16::<BigEndian>(section.block_states.non_air_blocks)?;
 
             let bits_per_block = section.block_states.bits_per_block;
-            data.write_u8(bits_per_block).await?;
+            data.write_u8(bits_per_block)?;
             // If bits_per_block is 0, the section is using the single-value palette format
             // If bits_per_block is greater than 0, the section is using the indirect palette format
             if bits_per_block > 0 {
                 // Write the palette
-                VarInt::new(section.block_states.palette.len() as i32)
-                    .write_async(&mut data)
-                    .await?;
+                VarInt::new(section.block_states.palette.len() as i32).write(&mut data)?;
                 for palette_entry in &section.block_states.palette {
-                    palette_entry.write_async(&mut data).await?;
+                    palette_entry.write(&mut data)?;
                 }
 
                 // Write the data
-                VarInt::new(section.block_states.data.len() as i32)
-                    .write_async(&mut data)
-                    .await?;
+                VarInt::new(section.block_states.data.len() as i32).write(&mut data)?;
                 for data_entry in &section.block_states.data {
-                    data.write_i64(*data_entry).await?;
+                    data.write_i64::<BigEndian>(*data_entry)?;
```

Comment:
Don't use byteorder, all primitives implement `NetEncode` trait. Use it.

Reply:
they compile to the same thing and it is easier to understand what is going on, since it's not writing to a socket stream, using netencode seems misleading. There isn't a performance difference and it makes it clearer that it's not writing straight to a socket.